File: 프로그래머스/lv3/43238. 입국심사/입국심사.py
def solution(n, times):
    times.sort()
    left = 0
    right = times[-1] * n
    answer = 0
    
    while(left<=right):
        mid = (left+right)//2
        
        finish = 0
        for time in times:
            finish += (mid//time)
        
        if(finish<n):
            left = mid+1
        else:
            answer = mid
            right = mid-1
    
   
    return answer

File: 프로그래머스/lv3/43238. 입국심사/입국심사.swift
import Foundation

func solution(_ n:Int, _ times:[Int]) -> Int64 {
    
    var times:[Int] = times.sorted()
    let cnt = times.count
    var _left:Int64 = 0
    var _right:Int64 = Int64(times[cnt-1]) * Int64(n)

    var ans:Int64 = 0
    while _left <= _right {

        let mid:Int64 = (_left+_right) / 2
        
        var success:Int = 0
        
        for time in times{
            success += (Int(mid)/time)
        }
        
        
        if success < n {
            _left = mid + 1 
        }
        
        else {
            ans = mid
            _right = mid - 1 
        }
        
    }
    
    return ans
}